src/PubSub/Queue.php

```diff
@@ -0,0 +1,274 @@
+        ]);
+
+        return $publish;
+    }
+
+    /**
+     * Push a new job onto the queue after a delay.
+     *
+     * @param  \DateTimeInterface|\DateInterval|int $delay
+     * @param  string|object                        $job
+     * @param  mixed                                $data
+     * @param  string                               $queue
+     * @return mixed
+     */
+    public function later($delay, $job, $data = '', $queue = null)
+    {
+        return $this->pushRaw(
+            $this->createPayload($job, $queue, $data),
+            $queue,
+            ['available_at' => (string) $this->availableAt($delay)]
+        );
+    }
+
+    /**
+     * @param $queue
+     * @return string
+     */
+    private function getClass($queue = null)
+    {
+        if (!$queue) {
+            return Config::get('laravel-jobs-plain.pubsub-plain.default-handler');
+        }
+        $queue = explode('/', $queue);
+        $queue = end($queue);
+
+        return (array_key_exists($queue, Config::get('laravel-jobs-plain.pubsub-plain.handlers')))
+            ? Config::get('laravel-jobs-plain.pubsub-plain.handlers')[$queue]
+            : Config::get('laravel-jobs-plain.pubsub-plain.default-handler');
+    }
+
+    /**
+     * Pop the next job off of the queue.
+     *
+     * @param  string $queue
+     * @return \Illuminate\Contracts\Queue\Job|null
+     */
+    public function pop($queue = null)
+    {
+        $topic = $this->getTopic($queue);
+
+        $subscription = $this->subscribeToTopic($topic);
+
+        $messages = $subscription->pull([
+            'returnImmediately' => true,
+            'maxMessages'       => 1,
+        ]);
+        if (!empty($messages) && count($messages) > 0) {
+            $class = $this->getClass($queue);
+
+            return new Job(
+                $this->container,
+                $this,
+                $messages[0],
+                $class,
+                $this->connectionName,
+                $queue
+            );
+        }
+    }
+
+    /**
+     * Push an array of jobs onto the queue.
+     *
+     * @param  array  $jobs
+     * @param  mixed  $data
+     * @param  string $queue
+     * @return mixed
+     */
+    public function bulk($jobs, $data = '', $queue = null)
+    {
+        $payloads = [];
+        foreach ((array) $jobs as $job) {
+            $payloads[] = ['data' => $this->createPayload($job, $queue, $data)];
+        }
+        $topic = $this->getTopic($queue);
+        $this->subscribeToTopic($topic);
+
+        return $topic->publishBatch($payloads);
+    }
+
+    /**
+     * Acknowledge a message.
+     *
+     * @param  \Google\Cloud\PubSub\Message $message
+     * @param  string                       $queue
+     */
+    public function acknowledge(Message $message, $queue = null)
+    {
+        $topic        = $this->getTopic($queue);
+        $subscription = $topic->subscription($this->getSubscriberName($topic));
+        $subscription->acknowledge($message);
```

Comment:
`$topic->subscription($this->getSubscriberName($topic))->acknowledge($message);`

AKA "memory might be cheap, it's not that cheap"